### DISCUSSION TITLE
Edge function with a query returns nothing

### DISCUSSION THREAD

**Discussion by fafa16** [supabase/supabase#22731] [Questions]
Hello eveyrone,
I am new to Edge function.I have done one to send push notification with onesignal and it works well.
the problem I have is that inside this Edge function, I wan to query first my profiles tables to get an information.
I don't knwo why but I can't have any result. Here is my code:
import { createClient } from 'https://esm.sh/@supabase/supabase-js@2.7.1'
import { FunctionsHttpError, FunctionsRelayError, FunctionsFetchError } from '@supabase/supabase-js'

Deno.serve(async (req) => {

const authHeader = req.headers.get('Authorization')!

  const supabase = createClient(
    Deno.env.get('SUPABASE_URL') ?? '',
    Deno.env.get('SUPABASE_ANON_KEY') ?? '',
    { global: { headers: { Authorization: authHeader } } }
  )

    const payload = await req.json();
    
    // Log the entire JSON payload
    console.log("Received JSON payload:", JSON.stringify(payload, null, 2));
  
    // Ensure that payload.record exists
    if (payload.record) {
      // Use payload.record.title and payload.record.content if present
      const title = payload.record.title || "Viibr";
      const content = payload.record.content ;
	  const userId = payload.record.user_id;
	  const campaign = payload.record.campaign || "PushNotification";
      const bigImage = payload.record['big-Image']  ;
		
	  let mess="MESSAGE";
	  
	  // Now we can get the session or user object
   
	   const { dataUser, error } = await supabase.from('profiles').select('*').eq('user_id',userId).maybeSingle();
	   console.log("Received from Supabase:", dataUser);
     
	   
			if (dataUser && dataUser.length > 0) {
			 mess = `Data received: ${JSON.stringify(dataUser)}`; // Utilisation de la donnée reçue
		} else if (error) {
			mess = `Error: ${error.message}`; // Utilisation du message d'erreur
		} else {
			mess = "No data received and no error reported.";
		}
     ...
     the end of the code works well (sending the notification) but I walways have mess = "No data received and no error reported.".
     dataUser  and error are always null
     
     Do I miss something??
     
     thank you in advance!!

**Comment by GaryAustin1** [comment]
dataUser will not work by itself is one issue.
`const {data:dataUser,error}` is probably what you want.

**Reply by fafa16** [reply]
Hi @GaryAustin1 ,
Thank you for the reply. I did the change but still the same issue...

**Reply by GaryAustin1** [reply]
You don't mention RLS, but that is usually silent on select.  
You could log out userId and your Authorization header to see if they are valid.

**Reply by fafa16** [reply]
RLS is very basic for now, I gave all access for now, no restrictions.
Authorization header looks good "Bearer myAnonApiKey" and userId is good also
I miss something...
Big thanks for the help!

**Reply by fafa16** [reply]
For Information, the edge function is called after an insert on a table:

create trigger "Push Notifications"
after insert on "Notifications2Push" for each row
execute function supabase_functions.http_request (
  'https://mylink.supabase.co/functions/v1/Send-Notifications',
  'POST',
  '{"Content-type":"application/json","Authorization":"Bearer XXX"}',
  '{}',
  '1000'
);

**Reply by GaryAustin1** [reply]
That row may not exist yet as the "transaction" for the insert is not done yet as you interrupted the process calling the function.
When your edge function issues a database call it is a brand new request not related to the process going on.
My guess anyway.

EDIT:  I was assuming the trigger was on the insert to the same table.  But now see it is a different table.  So maybe not.

**Reply by GaryAustin1** [reply]
Did you check your payload and user_id with console.logs?

**Reply by fafa16** [reply]
I confirm it is not the same table. the request is on 'profiles'. The payload is good, I have all the information. Do you know if I should have a specific RLS on profiles to allow a select query from the Edge function ?(I already do a lot of query from the app and it works)

**Reply by GaryAustin1** [reply]
I would test with RLS disabled for a  minute just to rule that out.
Otherwise no TO role or public will work with anon key and a TRUE USING policy for select.

**Reply by fafa16** [reply]
could you please explain this :"Otherwise no TO role or public will work with anon key and a TRUE USING policy for select.".
For now I only have  a "ALL" RLS =true (I aslo tested without RLS..)

**Reply by GaryAustin1** [reply]
```sql
create policy "Public profiles are visible to everyone."
on profiles for select
to public       
using ( true ); 
```
or 
```sql
create policy "Public profiles are visible to everyone."
on profiles for select
using ( true ); 
```
And without RLS means RLS disabled?

**Reply by GaryAustin1** [reply]
You also should probably show the current portion of your code with the data:dataUser change.

**Reply by fafa16** [reply]
![rls](https://github.com/supabase/supabase/assets/5648415/56ef9a9b-f9ad-409c-afdb-57ffcfe3dbf3)
and the code is:


	     const {data:dataUser, error } = await supabase.from('profiles').select('*').eq('user_id',userId).single();
	   console.log("Received from Supabase:", dataUser);
     
     
	   
		
			if (dataUser && dataUser.length > 0) {
			 mess = `Data received: ${JSON.stringify(dataUser)}`; // Utilisation de la donnée reçue
		} else if (error) {
			mess = `Error: ${error.message}`; // Utilisation du message d'erreur
		} else {
			mess = "No data received and no error reported.";
		}

**Reply by GaryAustin1** [reply]
If you drop your filter (.eq()) do you get an error about too many rows?
Unless your filter is bad,  I'm out of code ideas.

And your console.logs are sending back "Received from Supabase:  null" to the Edge logs?

Do you have more than one instance and maybe calling a profile table on another instance?

**Reply by fafa16** [reply]
yes if I drop the eq(), I have a "multiple or now rows returned".
Ye sI have a Received from Supabase: null in LOGS.
I have only one instance

**Reply by GaryAustin1** [reply]
Well either your userId is wrong being passed in, or it does not exist in the table when the select happens based on what you just said.  You got an error that multiple rows were returned which means your table was read.

**Reply by fafa16** [reply]
ok.if I test "const {data:dataUser, error } = await supabase.from('profiles').select('*');"  result is:
Log ID
f1be494f-8352-4a72-8c8d-8d996abd48f4

Log Timestamp (UTC)
2024-04-15T18:50:36.295Z

Log Event Message
Received from Supabase: [
  {
    created_at: "2024-03-04T10:32:49.163104+00:00",
    user_id: "b4378965-3439-422a-a4a1-6398d59c95f7",
    birthdate: "1999-12-08",
    gender: 2,
    name: "Caroline",
    id: "b4785b3f-adcb-4c9b-be4d-4458a0387c10",
    localisation: "0101000020E61000006DC5FEB27B720440454772F90F814840",
    height: 165,
    preferred_gender: 1,
    online: null
  },
  ]
  
  I guess my previous query has a bug...will investigate.
  
  
  
  Thanks again for the time spent!!!!!!
